client/src/context/useNotificationContext.tsx

```diff
@@ -0,0 +1,37 @@
+import { Notification } from '../interface/Notification';
+import { fetchNotifications } from '../helpers/APICalls/getNotifications';
+import { updateNotifications } from '../helpers/APICalls/updateNotifications';
+import { createContext, FunctionComponent, useState, useContext, useEffect } from 'react';
+
+interface INotificationContext {
+  notifications: [Notification] | null | undefined;
+  update: () => void;
+}
+
+export const NotificationContext = createContext<INotificationContext>({
+  notifications: null,
+  update: () => null,
+});
+
+export const NotificationProvider: FunctionComponent = ({ children }): JSX.Element => {
+  const [notifications, readNotifications] = useState<[Notification] | null | undefined>();
+
+  useEffect(() => {
+    const getNotifications = async () => {
+      const fetchedNotifications = await fetchNotifications();
+      readNotifications(fetchedNotifications);
+    };
+    getNotifications();
+  }, []);
+
+  const update = async () => {
+    const updated = await updateNotifications();
+    readNotifications(updated);
+  };
```

Comment:
Since we expose this in the context, let's rename it more descriptively? Maybe something like `markNotificationsAsRead`, what do you think?

Reply:
I agree, same comments as above about the word 'read': it needs more context. Change made.